### PR TITLE
[Admin] Locale unique code validation feature

### PR DESCRIPTION
**Pull request by Arminek**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

Based on #4594

### CODE REVIEW

File: features/locale/managing_locales/locale_unique_code_validation.feature [hunk 1]
```diff
@@ -5,12 +5,11 @@ Feature: Locale unique code validation
     I want to be prevented from adding a locale with an existing code
 
     Background:
-        Given the store is available in the Norwegian language
+        Given the store has locale "Norwegian"
```
Comment (michalmarcinkowski) [inline features/locale/managing_locales/locale_unique_code_validation.feature]: What was wrong with the previous step? :)

File: src/Sylius/Behat/spec/Context/Ui/Admin/ManagingLocalesContextSpec.php [hunk 2]
```diff
@@ -101,4 +101,11 @@ function it_throws_an_exception_if_resource_can_not_be_founded_on_page(IndexPage
 
         $this->shouldThrow(\InvalidArgumentException::class)->during('storeShouldBeAvailableInLanguage', ['Norwegian']);
     }
+
+    function it_throws_exception_if_locale_name_cannot_be_chosen_again(CreatePageInterface $createPage)
+    {
+        $createPage->chooseName('France')->willThrow(\Exception::class);
+
+        $this->shouldThrow(\Exception::class)->during('iShouldNotBeAbleToChoose', ['France']);
```
Comment (michalmarcinkowski) [inline src/Sylius/Behat/spec/Context/Ui/Admin/ManagingLocalesContextSpec.php]: Shouldn't this spec look like:
```php
$createPage->chooseName('France')->shouldBeCalled();

$this->shouldThrow(...)->during('iShouldNotBeAbleToChoose', ['France']);
```
?

Comment (Arminek) [inline src/Sylius/Behat/spec/Context/Ui/Admin/ManagingLocalesContextSpec.php]: `chooseName()` method won't throw exception cuz in spec we test our object isolated from UI so it doesn't know about presentation layer.

File: src/Sylius/Behat/Context/Ui/Admin/ManagingLocalesContext.php [hunk 3]
```diff
@@ -187,4 +188,12 @@ public function thisLocaleShouldBeDisabled(LocaleInterface $locale)
             sprintf('Locale %s should be disabled but it is not', $locale->getCode())
         );
     }
+
+    /**
+     * @Then I should not be able to choose :name
+     */
+    public function iShouldNotBeAbleToChoose($name)
+    {
+        expect($this->createPage)->toThrow(ElementNotFoundException::class)->during('chooseName', [$name]);
```
Comment (pjedrzejewski) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingLocalesContext.php]: We agreed to replace expect and catching exception with Assert and dedicated method on page that checks ``isSelectOptionAvailable`` or smth similar.

Comment (Arminek) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingLocalesContext.php]: `isOptionAvailable` ?

Comment (pjedrzejewski) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingLocalesContext.php]: :+1:

File: features/locale/managing_locales/locale_unique_code_validation.feature [hunk 4]
```diff
@@ -5,12 +5,10 @@ Feature: Locale unique code validation
...
         And I am logged in as an administrator
 
-    @todo
+    @ui
     Scenario: Trying to add new locale with used code
-        Given I want to create new locale
-        When I choose Norwegian
-        And I try to add it
-        Then I should be notified that it is not possible
+        Given I want to create a new locale
```
Comment (michalmarcinkowski) [inline features/locale/managing_locales/locale_unique_code_validation.feature]: to ~~create~~ **add**

**Comment by lchrusciel** [comment]
:+1:

File: src/Sylius/Behat/Context/Ui/Admin/ManagingLocalesContext.php [hunk 5]
```diff
@@ -66,6 +66,7 @@ public function __construct(
 
     /**
      * @Given I want to create a new locale
```
Comment (michalmarcinkowski) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingLocalesContext.php]: This should be removed and the below step should be used everywhere. Task for a separate PR.

**Comment by pjedrzejewski** [comment]
Thanks Arek!

**Pull request merged**